packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -134,21 +138,70 @@ type GraphiQLState = {
  *
  * @see https://github.com/graphql/graphiql#usage
  */
-export class GraphiQL extends React.Component<GraphiQLProps, GraphiQLState> {
-  /**
-   * Static Methods
-   */
-  static formatResult(result: any) {
-    return JSON.stringify(result, null, 2);
-  }
+export const GraphiQL: React.FC<GraphiQLProps> &
+  GraphiQLStaticProperties = props => {
+  return (
+    <MigrationContextProvider>
+      <GraphiQLInternals {...props} />
+    </MigrationContextProvider>
+  );
+};
 
-  static formatError(rawError: Error) {
-    const result = Array.isArray(rawError)
-      ? rawError.map(formatSingleError)
-      : formatSingleError(rawError);
-    return JSON.stringify(result, null, 2);
-  }
+interface GraphiQLStaticProperties {
+  formatResult: (result: any) => string;
+  formatError: (rawError: Error) => string;
+  Logo: typeof GraphiQLLogo;
+  Toolbar: typeof GraphiQLToolbar;
+  Footer: typeof GraphiQLFooter;
+  QueryEditor: typeof QueryEditor;
+  VariableEditor: typeof VariableEditor;
+  ResultViewer: typeof ResultViewer;
+  Button: typeof ToolbarButton;
+  ToolbarButton: typeof ToolbarButton;
+  Group: typeof ToolbarGroup;
+  Menu: typeof ToolbarMenu;
+  MenuItem: typeof ToolbarMenuItem;
+}
```

Comment:
Gotta be a better way to do this.

Reply:
we're about to drop this interface for 1.0.0 so don't worry about it too much:
https://github.com/graphql/graphiql/issues/1165